src/Laravel/Eloquent/Metadata/Factory/Property/EloquentPropertyNameCollectionMetadataFactory.php

```diff
@@ -39,8 +39,12 @@ public function create(string $resourceClass, array $options = []): PropertyName
             return $this->decorated?->create($resourceClass, $options) ?? new PropertyNameCollection();
         }
 
-        $refl = new \ReflectionClass($resourceClass);
         try {
+            $refl = new \ReflectionClass($resourceClass);
+            if ($refl->isAbstract()) {
+                return $this->decorated?->create($resourceClass, $options) ?? new PropertyNameCollection();
```

Comment:
Isn't this a bug fix as well that should land in 3.4?

Reply:
It's not inside 3.4 as it's a Laravel-related feature.